src/Evolution/Systems/GrMhd/ValenciaDivClean/PrimitiveFromConservative.cpp

```diff
@@ -124,6 +124,21 @@ bool PrimitiveFromConservative<OrderedListOfPrimitiveRecoverySchemes,
     // Quick exit from inversion in low-density regions where we will
     // apply atmosphere corrections anyways.
     if (rest_mass_density_times_lorentz_factor[s] < cutoffD) {
+      if constexpr (ThermodynamicDim == 3) {
+        const double specific_energy_at_point =
+            equation_of_state.specific_internal_energy_lower_bound(
+                floorD, get(*electron_fraction)[s]);
+        const double pressure_at_point =
+            get(equation_of_state.pressure_from_density_and_energy(
+                Scalar<double>{floorD},
+                Scalar<double>{specific_energy_at_point},
+                Scalar<double>{get(*electron_fraction)[s]}));
+        const double specific_enthalpy_at_point =
+            1.0 + specific_energy_at_point + pressure_at_point / floorD;
+        primitive_data = PrimitiveRecoverySchemes::PrimitiveRecoveryData{
+            floorD, 1.0, pressure_at_point, floorD * specific_enthalpy_at_point,
+            get(*electron_fraction)[s]};
+      }
```

Comment:
`else`

Also this file needs to be run through clang-format. If you do, please commit separately.